server/src/main/scala/controllers/application/exports/WithDesksExport.scala

```diff
@@ -1,44 +1,165 @@
+                                                localDate: String,
+                                                pointInTime: String,
+                                                terminalName: String): Action[AnyContent] =
+    authByRole(DesksAndQueuesView) {
 
-  def exportDesksAndQueuesAtPointInTimeCSV(pointInTime: String,
-                                           terminalName: String): Action[AnyContent] =
-    authByRole(DesksAndQueuesView)(exportPointInTimeView(terminalName, pointInTime))
+      (LocalDate.parse(localDate), Try(SDate(pointInTime.toLong))) match {
+        case (Some(ld), Success(pit)) =>
 
-  def exportDesksAndQueuesBetweenTimeStampsCSV(startMillis: String,
-                                               endMillis: String,
-                                               terminalName: String): Action[AnyContent] =
-    authByRole(DesksAndQueuesView)(exportEndOfDayView(startMillis, endMillis, terminalName))
+          val viewDay = SDate(ld)
+          val start = viewDay
+          val end = viewDay.getLocalNextMidnight.addMinutes(-1)
 
-  private def exportEndOfDayView(startMillis: String, endMillis: String, terminalName: String): Action[AnyContent] = {
-    val start = SDate(startMillis.toLong)
-    val end = SDate(endMillis.toLong)
-    val summaryForPeriodFn = Exports.queueSummariesFromPortState(airportConfig.nonTransferQueues(terminal(terminalName)), 15, Terminal(terminalName), queryFromPortStateFn(None))
-    Action(exportToCsv(start, end, "desks and queues", terminal(terminalName), Option((summaryActorProvider, GetSummaries)), summaryForPeriodFn))
-  }
+          exportStreamingDesksAndQueuesBetweenTimestampsCSV(
+            start,
+            end,
+            terminalName,
+            deskRecsExportStreamForTerminalDates(pointInTime = Option(pit.millisSinceEpoch)),
+            s"desks-and-queues-recs-at-${pit.toISOString()}-for"
+          )
+        case _ =>
+          Action(BadRequest(write(ErrorResponse("Invalid date format"))))
+      }
+    }
+
+  def exportDesksAndQueuesRecsBetweenTimeStampsCSV(startLocalDate: String,
+                                                   endLocalDate: String,
+                                                   terminalName: String): Action[AnyContent] =
+    authByRole(DesksAndQueuesView) {
+      (LocalDate.parse(startLocalDate), LocalDate.parse(endLocalDate)) match {
+        case (Some(startLD), Some(endLD)) =>
+
+          val start = SDate(startLD)
+          val end = SDate(endLD).getLocalNextMidnight.addMinutes(-1)
+
+          exportStreamingDesksAndQueuesBetweenTimestampsCSV(
+            start,
+            end,
+            terminalName,
+            deskRecsExportStreamForTerminalDates(pointInTime = None),
+            "desks-and-queues-recs"
+          )
+        case _ =>
+          Action(BadRequest(write(ErrorResponse("Invalid date format"))))
+      }
+    }
+
+  def exportDesksAndQueuesDepsAtPointInTimeCSV(
```

Comment:
Since this function and its desk recs counterpart `exportDesksAndQueuesRecsAtPointInTimeCSV` are doing the same thing other than calling different functions, it could make sense to unify them by adding the function as a parameter

Reply:
I've tried various forms of doing this, and I think it makes things harder to comprehend rather than easier, there's probably a simplification available here, but it's complicated by the fact that we are currently fudging the point in time export date range to be the day of the point in time - there are two slightly different use cases here for the 4 different exports.

Reply:
👍